### PR TITLE
Move transaction statistics to a separate microservice - Closes #999

### LINKED ISSUE #999: Move transaction statistics to a separate microservice
### Description

Move the transaction statistics feature to a separate microservice.
Use the index to create aggregated results.

The statistics generation is a computational-intensive operation that should be handled by a separate process.

Service name: `statistics`.

The methods are kept the same.

- 'transactions.statistics.day'
- 'transactions.statistics.month'

### Acceptance Criteria

- The statistics are generated within the same boundaries as before (1 year)
- Keep data store abstraction
- Updated unit tests
- Updated functional tests

Consider the proposed architecture.

Lisk Service-Planned arch drawio (7)

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?
This PR resolves #999 

### How was it solved?
- [x] Create new service `transaction-statistics`
- [x] Refactor/Migrate code:
  - [x] Methods
  - [x] Current implementation of statistics
  - [x] Mysql
- [x] Use data service to retrieve transactions
- [x] Fix for services dependency (wait for service until available)
- [x] Add logic to Init stats only after block indexing is complete

### How was it tested?
Local

### CODE REVIEW

**Review by sameersubudhi — COMMENTED** [review]
Please update the Makefile to include `blockchain-coordinator` and `transaction-statistics` microservices.

**Comment by nagdahimanshu** [comment]
> Please update the Makefile to include `blockchain-coordinator` and `transaction-statistics` microservices.

Done

File: services/transaction-statistics/config.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+config.brokerTimeout = Number(process.env.SERVICE_BROKER_TIMEOUT) || 10; // in seconds
+config.httpTimeout = Number(process.env.LISK_CORE_CLIENT_TIMEOUT) || 30; // in seconds
+
+/**
+ * External endpoints
+ */
+config.endpoints.mysql = process.env.SERVICE_CORE_MYSQL || 'mysql://lisk:password@localhost:3306/lisk';
+
+config.transactionStatistics = {
+	historyLengthDays: Number(process.env.TRANSACTION_STATS_HISTORY_LENGTH_DAYS || 365),
```
Comment (sameersubudhi) [inline services/transaction-statistics/config.js:34]:
```suggestion
	historyLengthDays: Number(process.env.TRANSACTION_STATS_HISTORY_LENGTH_DAYS || 366),
```

File: services/transaction-statistics/config.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+config.debug = process.env.SERVICE_LOG_LEVEL === 'debug';
+
+config.queue = {
+	defaultJobOptions: {
+		attempts: 5,
+		timeout: 5 * 60 * 1000, // millisecs
+		removeOnComplete: true,
+	},
+	settings: {},
+	// limiter: {},
```
Comment (sameersubudhi) [inline services/transaction-statistics/config.js:61]: Please remove them if they're not used anywhere.

File: services/transaction-statistics/shared/transactionStatistics.js [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+	}
+};
+
+const queueName = 'transactionStats';
+const transactionStatisticsQueue = Queue(
+	config.endpoints.redis,
+	queueName,
+	queueJob,
+	1,
+	config.Queue.defaultJobOptions,
```
Comment (sameersubudhi) [inline services/transaction-statistics/shared/transactionStatistics.js:175]:
```suggestion
	config.queue.defaultJobOptions,
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]
Please consider the following. Also, there are many logs with the following errors:
`2022-04-22T12:26:16.605 ERROR [mysql] ER_CON_COUNT_ERROR: Too many connections`.

**Comment by nagdahimanshu** [comment]
> Please consider the following. Also, there are many logs with the following errors:

@sameersubudhi I updated the MySql docker-compose file to increase the default connection limit to 200. Can you try again by pulling the latest changes?

**Review by sameersubudhi — CHANGES_REQUESTED** [review]
Please add the logic to start computing the stats only after the indexer sends out an event confirming that the index is completely up-to-date.

File: services/transaction-statistics/config.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+	endpoints: {},
+	log: {},
+};
+
+/**
+ * Inter-service message broker
+ */
+config.transporter = process.env.SERVICE_BROKER || 'redis://localhost:6379/0';
+config.brokerTimeout = Number(process.env.SERVICE_BROKER_TIMEOUT) || 10; // in seconds
+config.httpTimeout = Number(process.env.LISK_CORE_CLIENT_TIMEOUT) || 30; // in seconds
```
Comment (sameersubudhi) [inline services/transaction-statistics/config.js:26]: This config property is getting used anywhere, please remove it if it is unnecessary.

**Review by sameersubudhi — APPROVED** [review]
Thank you for the changes. Great job! 🎉

**Pull request merged**